client/state/plugins/marketplace/reducer.ts

```diff
@@ -40,19 +50,66 @@ export function purchaseFlow(
 			action = action as ISetPluginToBeInstalledAction;
 			return {
 				...state,
+				siteTransferStatus: MARKETPLACE_ASYNC_PROCESS_STATUS.UNKNOWN,
+				pluginInstallationStatus: MARKETPLACE_ASYNC_PROCESS_STATUS.UNKNOWN,
 				pluginSlugToBeInstalled: action.pluginSlugToBeInstalled,
 			};
 		case MARKETPLACE_PLUGIN_INSTALLED_ON_PURCHASE:
 			action = action as ISetPluginInstalledDuringPurchaseFlag;
 			return {
 				...state,
+				siteTransferStatus: MARKETPLACE_ASYNC_PROCESS_STATUS.UNKNOWN,
+				pluginInstallationStatus: MARKETPLACE_ASYNC_PROCESS_STATUS.UNKNOWN,
 				isPluginInstalledDuringPurchase: action.isPluginInstalledDuringPurchase,
 			};
+		case THEME_TRANSFER_INITIATE_REQUEST:
+			return {
+				...state,
+				siteTransferStatus: MARKETPLACE_ASYNC_PROCESS_STATUS.IN_PROGRESS,
```

Comment:
The lack of tests for these reducer changes is disturbing. Being pure functions, reducers are among the most straightforward functionality to write unit tests for, and unit tests have proven to be valuable for them. Unit tests seem particularly valuable for this reducer which has grown to be bigger. Have you considered adding some?

Reply:
Yes I will add them right away. Was caught off guard with the tough functionality. I will write some test after finishing these review fixes 👍

Reply:
I will pick up these tests in the next PR, right after this 👍